src/s390_ecc.c

```diff
@@ -2468,55 +2468,124 @@ unsigned int eckeygen_sw(ICA_EC_KEY *key)
 	return rc;
 }
 
-/**
- * Check if the given ICA_EC_KEY is valid. This check is performed via openssl,
- * so we can only check keys that are supported by openssl. If e.g. openssl
- * is in fips mode, very few curves are supported.
- *
- * @return:
- *    0       success
- *    EINVAL  key check failed
+/*
+ * Check if the given pkey has the same public key (X,Y) as the given
+ * (X,Y) values.
  */
-int ec_key_check(const ICA_EC_KEY *ica_key)
+int pubkey_match(EVP_PKEY *pkey, unsigned char *X, unsigned char *Y)
 {
-	EVP_PKEY *privkey = NULL, *pubkey = NULL;
-	BIGNUM *d = NULL, *x = NULL, *y = NULL;
-	int privlen, rc = EINVAL;
+	EC_KEY *eckey;
+	EC_POINT *point;
+	unsigned char pubbuf[133];
+	size_t publen;
+	int match = 0;
+
+	eckey = EVP_PKEY_get1_EC_KEY(pkey);
```

Comment:
`EVP_PKEY_get1_EC_KEY()` is deprectated with OpenSSL 3.0.
`Same for EC_KEY_get0_public_key()`.
Use `EVP_PKEY_todata()` for the OpenSSL 3.0 case.

Reply:
OK, after our internal discussion I'm removing the pubkey_match() routine, which also removes the use of deprecated openssl functions.